src/components/ExperienceForm/ExperienceForm.js

```diff
@@ -3,155 +3,222 @@
  * Exports a component that allows users to operate on experiences.
  */
 
-import React, { Component } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
+import { withFormik } from 'formik';
+import { string, object } from 'yup';
 import { TextField, Button, withStyles } from '@material-ui/core';
 
 import { Message } from '../';
 import ExperienceFormStyles from './ExperienceForm.style';
+import parseExperience from '../../lib/parseExperience';
 import {
   EXPERIENCES_CREATE,
   EXPERIENCES_EDIT,
-  FORM_BUTTON_INSERT_UPDATE
+  FORM_BUTTON_INSERT_UPDATE,
+  FORM_INPUT_VALIDATION_REGEX_URL_SEGMENT
 } from '../../constants';
 
-class ExperienceForm extends Component {
-  static propTypes = {
-    submitHandler: PropTypes.oneOfType([PropTypes.func, PropTypes.bool]),
-    experienceSlug: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
-    classes: PropTypes.shape({
-      textField: PropTypes.string.isRequired,
-      button: PropTypes.string.isRequired
-    }).isRequired,
-    dispatch: PropTypes.func.isRequired,
-    history: PropTypes.shape({
-      push: PropTypes.func.isRequired
-    }).isRequired,
-    user: PropTypes.shape({
-      authentication: PropTypes.shape({
-        accessToken: PropTypes.string.isRequired,
-        csrfToken: PropTypes.string.isRequired
-      }).isRequired
-    }).isRequired,
-    experiences: PropTypes.shape({
-      error: PropTypes.string,
-      items: PropTypes.arrayOf(
-        PropTypes.shape({
-          id: PropTypes.string.isRequired,
-          title: PropTypes.string.isRequired,
-          body: PropTypes.PropTypes.shape({
-            value: PropTypes.string
-          }),
-          field_experience_path: PropTypes.string.isRequired
-        })
-      )
-    })
-  };
+const ExperienceForm = ({
+  classes,
+  experiences: { error: apiError },
+  user: { username },
+  values,
+  errors,
+  touched,
+  isSubmitting,
+  handleSubmit,
+  handleChange,
+  handleBlur
+}) => (
+  <form onSubmit={handleSubmit}>
+    {apiError && <Message>{apiError}</Message>}
+    <TextField
+      id="title"
+      label="Title"
+      type="text"
+      required
+      helperText={
+        errors.title
+          ? errors.title
+          : 'Enter a user-friendly title for your experience.'
+      }
+      value={values.title}
+      onChange={handleChange}
+      onBlur={handleBlur}
+      error={!!errors.title && touched.title}
```

Comment:
MaterialUI exposes a nice `error` property that takes a boolean, which indicates whether or not the form field should have a red border, and red helper text. Nice!

`!!errors.title && touched.title` will be `true` only if the title field has been modified since initialization, and there is an error for the title field